src/libAtomVM/externalterm.c

```diff
@@ -488,12 +508,18 @@ static term parse_external_terms(const uint8_t *external_term_buf, int *eterm_si
     }
 }
 
-static int calculate_heap_usage(const uint8_t *external_term_buf, int *eterm_size, bool copy, Context *ctx)
+static int calculate_heap_usage(const uint8_t *external_term_buf, size_t remaining, int *eterm_size, bool copy, Context *ctx)
 {
+    if (remaining < 1) {
```

Comment:
Let's add UNLIKELY to error handling branches.
My assumption is that errors are less frequent.

I will not make this comment for all the other places where we may add UNLIKELY.

Reply:
I think I got them all.